src/main/java/subway/line/LineResponse.java

```diff
@@ -23,4 +25,14 @@ public LineResponse(Line line, List<LineStationsResponse> stations) {
         this.color = line.getColor();
         this.stations = stations;
     }
+
+    public static LineResponse from(Line line) {
+        List<LineStationsResponse> stationList = line.getSections().stream()
+                .flatMap(section -> Stream.of(new LineStationsResponse(section.getUpStation()),
+                        new LineStationsResponse(section.getDownStation())))
+                .distinct()
+                .collect(Collectors.toList());
```

Comment:
직접 구간정보 목록을 꺼내서 작업하는 방식도 있지만 저는 고차함수를 이용한 방식도 좋아하는 편인데 고려해봐도 좋을 것 같아요.

```suggestion
        List<LineStationsResponse> stationList = line.mapSectionStations(LineStationsResponse::new)
```

Reply:
말씀하신 부분이 DTO 객체에 엔티티를 DTO로 변환시킬 메서드를 작성할 것인가, 아니면 엔티티 객체에 엔티티를 DTO로 변환시킬 메서드를 작성할 것인가를 선택해야 하는 문제가 아닌가 생각이 들었습니다.

우선 저렇게 코드를 작성하게 된 저의 생각을 조심스럽게 공유드리겠습니다.
1. 변화의 관점에서 LineResponse 응답의 형태가 바뀐다면 기존에 작성한 부분만 변경하면 될거라고 생각했습니다.
- Line에 존재하는 데이터를 조작해서 응답의 형태로 맞춰주기만 하면 된다고 느껴서 LineResponse 클래스만 수정하면 되지 않을까 생각했습니다.


2. 이 이유가 좀 더 큰데 Line 객체가 핵심 도메인 엔티티라고 생각했기 때문에 DTO 변환과 같은 로직을 엔티티 내부에 작성하고 싶지 않았습니다.
- 핵심 도메인 엔티티에서는 비즈니스와 관련된 메서드가 주로 오는게 좋을거라고 생각했습니다.
- 현재는 LineResponse DTO 한 개이지만 이후에 여러 DTO가 추가될때마다 도메인 엔티티에 변환 로직이 들어가는 것보다는 각 DTO 클래스에서 엔티티를 의존해서 변환하는 로직을 갖는게 좋지 않을까?? 라는 생각을 했습니다.


제가 현업 경험이 많지 않아서 실제 현업에서 어떻게 주로 사용하는지 잘 모르다보니 개인의 주관이 좀 많이 들어간 부분이 있는것 같습니다. 리뷰어님의 경험을 바탕으로 어떤식으로 처리한다고 경험을 공유해주시면 많은 도움이 될 것 같습니다!!

코드는 말씀하신 대로 고차함수를 이용해 추상화 시키도록 변경했습니다!!
감사합니다~~

Reply:
아 코드를 수정하다보니 제가 답변을 좀 잘못 이해한것 같습니다.
고차 함수가 결국 추상화 해서 DTO를 변환하는 부분을 Line 엔티티에 두는 것으로 이해를 했는데

함수형 인터페이스를 활용하라는 의도로 피드백을 주신것 같습니다.
피드백 주신 내용 이해를 좀 잘못 한것 같은데, 그래도 혹시 위 질문과 겹치는 부분이 있는것 같아서 답변 주시면 감사하겠습니다!!

코드는 고차함수를 이용하도록 변경했습니다!!
감사합니다!!

Reply:
네 두 번째 댓글처럼, 제 의견은 엔티티에 dto를 의존하는 로직이 있어야한다가 아니라 ㅎㅎ 행위 자체를 전달하여 동작하게 함으로써 엔티티는 별도의 dto 의존성을 두지 않으면서 내부 값을 외부로 다 꺼내어 제공하는것도 최소화 하면 좋겠다는 취지의 피드백이였습니다 ㅎㅎ